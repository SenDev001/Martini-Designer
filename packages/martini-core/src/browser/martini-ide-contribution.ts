import {
    Command,
    CommandContribution,
    CommandRegistry,
    MAIN_MENU_BAR,
    MenuContribution,
    MenuModelRegistry
} from "@theia/core";
import {
    ApplicationShell,
    CommonCommands,
    KeybindingContribution,
    KeybindingRegistry,
    OpenerService
} from "@theia/core/lib/browser";
import { inject, injectable } from "inversify";
import messages from "martini-messages/lib/messages";
import { HistoryManagerProvider } from "./history/history-manager";
import URI from "@theia/core/lib/common/uri";

export const DeleteCommand: Command = {
    id: "martini.delete",
    label: messages.delete,
    iconClass: "martini-icon martini-delete-icon"
};

export const RenameCommand: Command = {
    id: "martini.rename",
    label: messages.rename,
    iconClass: "martini-icon martini-edit-icon"
};

export const EditCommand: Command = {
    id: "martini.edit",
    label: messages.edit,
    iconClass: "martini-icon martini-edit-icon"
};

export const CopyPathCommand: Command = {
    id: "martini.copyPath",
    label: messages.copy_path,
    iconClass: "martini-icon martini-copy-icon"
};

export const CopyNamespaceCommand: Command = {
    id: "martini.copyNamespace",
    label: messages.copy_namespace,
    iconClass: "martini-icon martini-copy-icon"
};

export type MoveDirection = "left" | "right" | "up" | "down";

export const MoveLeftCommand: Command = {
    id: "martini.moveLeft",
    label: messages.move_left,
    iconClass: "martini-icon martini-arrow-left-icon"
};

export const MoveRightCommand: Command = {
    id: "martini.moveRight",
    label: messages.move_right,
    iconClass: "martini-icon martini-arrow-right-icon"
};

export const MoveUpCommand: Command = {
    id: "martini.moveUp",
    label: messages.move_up,
    iconClass: "martini-icon martini-arrow-up-icon"
};

export const MoveDownCommand: Command = {
    id: "martini.moveDown",
    label: messages.move_down,
    iconClass: "martini-icon martini-arrow-down-icon"
};

export const EditMartiniPropertiesCommand: Command = {
    id: "martini.editMartiniProperties",
    label: messages.edit_martini_properties,
    iconClass: "martini-icon martini-edit-icon"
};

export const EditComments: Command = {
    id: "martini.editComments",
    label: messages.edit_comments,
    iconClass: "martini-icon martini-comments-icon"
};

export const ToggleEnabled: Command = {
    id: "martini.toggleEnabled",
    label: messages.toggle_enabled,
    iconClass: "martini-icon martini-disabled-icon"
};

export namespace MartiniMenus {
    export const MARTINI = [...MAIN_MENU_BAR, "0_martini"];
}

@injectable()
export class MartiniIdeCommandContribution implements CommandContribution {
    @inject(ApplicationShell)
    private readonly shell: ApplicationShell;
    @inject(OpenerService)
    private readonly openerService: OpenerService;

    registerCommands(commands: CommandRegistry): void {
        commands.registerCommand(DeleteCommand);
        commands.registerCommand(RenameCommand);
        commands.registerCommand(CopyPathCommand);
        commands.registerCommand(CopyNamespaceCommand);
        commands.registerCommand(MoveLeftCommand);
        commands.registerCommand(MoveRightCommand);
        commands.registerCommand(MoveUpCommand);
        commands.registerCommand(MoveDownCommand);
        commands.registerCommand(EditComments);
        commands.registerCommand(ToggleEnabled);
        commands.registerCommand(EditCommand);
        CommonCommands.COPY.iconClass = "martini-icon martini-copy-icon";
        CommonCommands.CUT.iconClass = "martini-icon martini-cut-icon";
        CommonCommands.PASTE.iconClass = "martini-icon martini-paste-icon";
        CommonCommands.FIND.iconClass = "martini-icon martini-search-icon";

        const isTextElementFocused = () => {
            const activeElement = document.activeElement;
            return (
                (activeElement instanceof HTMLInputElement && activeElement.type === "text") ||
                activeElement instanceof HTMLTextAreaElement
            );
        };
        commands.registerHandler(CommonCommands.UNDO.id, {
            execute: () => {
                document.execCommand("undo", true);
            },
            isEnabled: isTextElementFocused
        });
        commands.registerHandler(CommonCommands.REDO.id, {
            execute: () => {
                document.execCommand("redo", true);
            },
            isEnabled: isTextElementFocused
        });
        commands.registerHandler(CommonCommands.UNDO.id, {
            execute: () => {
                const widget = this.shell.currentWidget;
                return HistoryManagerProvider.is(widget) && widget.historyManager?.undo();
            },
            isEnabled: () => {
                const widget = this.shell.currentWidget;
                return (
                    HistoryManagerProvider.is(widget) &&
                    widget.historyManager !== undefined &&
                    widget.historyManager.canUndo()
                );
            }
        });
        commands.registerHandler(CommonCommands.REDO.id, {
            execute: () => {
                const widget = this.shell.currentWidget;
                return HistoryManagerProvider.is(widget) && widget.historyManager?.redo();
            },
            isEnabled: () => {
                const widget = this.shell.currentWidget;
                return (
                    HistoryManagerProvider.is(widget) &&
                    widget.historyManager !== undefined &&
                    widget.historyManager.canRedo()
                );
            }
        });
        commands.registerCommand(EditMartiniPropertiesCommand, {
            execute: async () => {
                const uri: URI = new URI("martini://core/conf/properties/application.properties");
                const handler = await this.openerService.getOpener(uri);
                return handler.open(uri);
            }
        });
    }
}

@injectable()
export class MartiniMenuContribution implements MenuContribution {
    registerMenus(menus: MenuModelRegistry): void {
        menus.registerSubmenu(MartiniMenus.MARTINI, "Martini");
        menus.registerMenuAction(MartiniMenus.MARTINI, {
            commandId: EditMartiniPropertiesCommand.id
        });
    }
}

@injectable()
export class MartiniIdeKeybindingContribution implements KeybindingContribution {
    registerKeybindings(keybindings: KeybindingRegistry): void {
        keybindings.registerKeybindings(
            {
                command: DeleteCommand.id,
                keybinding: "del"
            },
            {
                command: RenameCommand.id,
                keybinding: "f2"
            },
            {
                command: EditCommand.id,
                keybinding: "ctrlcmd+enter"
            },
            {
                command: CopyPathCommand.id,
                keybinding: "ctrlcmd+shift+C"
            },
            {
                command: CopyNamespaceCommand.id,
                keybinding: "ctrlcmd+alt+shift+C"
            },
            {
                command: CommonCommands.PASTE.id,
                keybinding: "ctrlcmd+V"
            },
            {
                command: MoveLeftCommand.id,
                keybinding: "ctrlcmd+left"
            },
            {
                command: MoveRightCommand.id,
                keybinding: "ctrlcmd+right"
            },
            {
                command: MoveUpCommand.id,
                keybinding: "ctrlcmd+up"
            },
            {
                command: MoveDownCommand.id,
                keybinding: "ctrlcmd+down"
            },
            {
                command: EditComments.id,
                keybinding: "ctrlcmd+/"
            },
            {
                command: ToggleEnabled.id,
                keybinding: "alt+/"
            }
        );
    }
}
