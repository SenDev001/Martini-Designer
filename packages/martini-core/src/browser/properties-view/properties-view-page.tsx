import { Widget } from "@phosphor/widgets";
import { Emitter, Event } from "@theia/core";
import { injectable } from "inversify";
import * as React from "react";
import styled from "styled-components";
import messages from "martini-messages/lib/messages";

/**
 * Represents a page to be rendered in the PropertiesViewWidget.
 */
export interface PropertiesViewPage {
    onChange: Event<void>;
    /**
     * Whether or not the page should be scrolled vertically when the content overflows.
     */
    scrollY: boolean;

    /**
     * Called when the selection changed.
     * @param source the widget source of the selection
     * @param selection the current selection
     */
    onSelectionChange(source: Widget, selection: object | undefined): void;

    /**
     * Renders the page.
     */
    render(): React.ReactNode;
}

/**
 * Provides the page to be rendered for the current widget.
 */
export interface PropertiesViewPageProvider {
    getPropertiesViewPage(): PropertiesViewPage;
}

export namespace PropertiesViewPageProvider {
    export function is(object: any): object is PropertiesViewPageProvider {
        return !!object && "getPropertiesViewPage" in object;
    }
}

interface PageProps {
    scrollY: boolean;
}

const Page = styled.div<PageProps>`
    display: grid;
    grid-template-rows: max-content 1fr;
    grid-row-gap: 5px;
    height: 100%;
    overflow-y: ${ props => props.scrollY ? "auto" : undefined};

    .content {
        height: 100%;
        overflow-y: hidden;
    }
`;

const TitleBar = styled.div`
    display: grid;
    margin-left: 4px;
    grid-template-columns: max-content 1fr ${props => `repeat(${React.Children.count(props.children) - 2}, max-content)`};
    grid-column-gap: 4px;

    .label {
        align-self: center;
    }

    .read-only {
        display: grid;
        grid-template-columns: max-content max-content;

        div {
            align-self: center;
        }
    }
`;

/**
 * Base implementation of a PropertiesViewPage, it has a title with icon and a toolbar.
 */
@injectable()
export abstract class AbstractPropertiesViewPage implements PropertiesViewPage {
    protected readonly onChangeEmitter = new Emitter<void>();
    readonly onChange = this.onChangeEmitter.event;
    protected readOnly = false;
    title: string | undefined = undefined;
    iconClass: string | undefined = undefined;
    scrollY = true;

    abstract onSelectionChange(source: Widget, selection: object): void;

    render(): React.ReactNode {
        return <Page scrollY={this.scrollY}>
            {this.title &&
                <TitleBar>
                    <div className={`martini-tree-icon ${this.iconClass}`} />
                    <div className="label">{this.title}</div>
                    <div className="label read-only">{!this.readOnly ? "" :
                        (<>
                            <div className="martini-tree-icon martini-lock-icon"/>
                            <div>({messages.read_only})</div>
                        </>)}
                    </div>
                    {this.renderToolBar()}
                </TitleBar>
            }
            <div className="content">
                {this.doRender()}
            </div>
        </Page>;
    }

    protected abstract doRender(): React.ReactNode;

    protected renderToolBar(): React.ReactNode {
        return undefined;
    }
}
