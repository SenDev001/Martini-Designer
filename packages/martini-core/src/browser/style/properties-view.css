.properties-table {
    padding: 0;
    width: 100%;
}

.properties-table table {
    border-spacing: 0;
    width: 100%;
}

.properties-table tr:last-child td {
    border-bottom: 0;
}

.properties-table th {
    margin: 0;
    padding: 0.5rem;
    border-bottom: 1px solid var(--martini-table-border);
    border-right: 1px solid var(--martini-table-border);
}

.properties-table td {
    margin: 0;
    height: 25px;
    padding: 0 5px 0px 4px;
    border-bottom: 1px solid var(--martini-table-border);
    border-right: 1px solid var(--martini-table-border);
}

.properties-table th:last-child,
td:last-child {
    border-right: 0;
}