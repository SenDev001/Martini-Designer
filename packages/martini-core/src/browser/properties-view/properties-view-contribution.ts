import { Command } from "@theia/core";
import { AbstractViewContribution } from "@theia/core/lib/browser";
import messages from "martini-messages/lib/messages";
import { PropertiesViewWidget } from "./properties-view-widget";

export const TogglePropertiesViewCommand: Command = {
    id: "martini.togglePropertiesView",
    label: messages.toggle_props_view
};

export class PropertiesViewContribution extends AbstractViewContribution<PropertiesViewWidget> {
    constructor() {
        super({
            widgetId: PropertiesViewWidget.ID,
            widgetName: messages.properties,
            toggleKeybinding: "ctrlcmd+shift+O",
            toggleCommandId: TogglePropertiesViewCommand.id,
            defaultWidgetOptions: {
                area: "right"
            }
        });
    }
}
