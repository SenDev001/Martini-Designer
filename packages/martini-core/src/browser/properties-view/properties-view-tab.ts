import { PropertiesViewPage } from "./properties-view-page";

export const PropertiesViewTab = Symbol("PropertiesViewTab");

/**
 * Represents a tab to rendered in a TabbedPropertiesViewPage.
 */
export interface PropertiesViewTab extends PropertiesViewPage {
    /**
     * The name of the tab.
     */
    name: string;

    /**
     * The order of the tab, ordered from lowest to highest value for left to right tab.
     */
    order: number;

    /**
     * Returns whether or not this tab should be visible, based on the given selection.
     * @param selection the current selection
     */
    isVisible(selection: object): boolean;
}
