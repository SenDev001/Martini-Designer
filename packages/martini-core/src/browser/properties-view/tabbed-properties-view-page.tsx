import { Widget } from "@phosphor/widgets";
import { DisposableCollection } from "@theia/core";
import { injectable, multiInject } from "inversify";
import * as React from "react";
import { Tab, TabList, TabPanel, Tabs } from "react-tabs";
import { AbstractPropertiesViewPage } from "./properties-view-page";
import { PropertiesViewTab } from "./properties-view-tab";
import { NoPropertiesLabel } from "./properties-view-widget";

/**
 * PropertiesViewWidget page which can display other pages in tabs. To add a tab to this page, one has to bind a
 * PropertiesViewTab.
 */
@injectable()
export class TabbedPropertiesViewPage extends AbstractPropertiesViewPage {

    @multiInject(PropertiesViewTab)
    private tabs: PropertiesViewTab[];
    private visibleTabs: PropertiesViewTab[] = [];
    private tabDisposables = new DisposableCollection();
    title = undefined;
    iconClass = undefined;
    protected selectedTabIndex: number = 0;
    scrollY = false;

    onSelectionChange(source: Widget, selection: object): void {
        this.tabDisposables.dispose();
        this.visibleTabs = this.tabs
            .filter(tab => tab.isVisible(selection))
            .sort((tab1, tab2) => tab1.order - tab2.order);
        if (this.selectedTabIndex >= this.visibleTabs.length)
            this.selectedTabIndex = 0;
        this.visibleTabs.forEach((tab, i) => {
            tab.onSelectionChange(source, selection);
            this.tabDisposables.push(tab.onChange(() => {
                // Only re render the active tab
                if (this.selectedTabIndex === i)
                    this.onChangeEmitter.fire();
            }));
        });
    }

    render(): React.ReactNode {
        if (this.visibleTabs.length === 0)
            return <NoPropertiesLabel />;

        return <Tabs
            onKeyPress={e => this.handleKeyPressed(e)}
            style={{
                display: "flex",
                flexDirection: "column",
                height: "100%"
            }}
            selectedIndex={this.selectedTabIndex}
            onSelect={(index => this.handleSelect(index))}
            forceRenderTabPanel={true}
        >
            {this.visibleTabs.map(tab => (
                <TabPanel key={tab.name} style={{
                    margin: 5,
                    flex: 1,
                    overflowY: "auto"
                }}>
                    {tab.render()}
                </TabPanel>))}
            <TabList style={{ flexShrink: 0 }}>
                {this.visibleTabs.map(tab => (<Tab key={tab.name}>{tab.name}</Tab>))}
            </TabList>
        </Tabs>;
    }

    doRender(): React.ReactNode {
        return undefined;
    }

    private handleKeyPressed(e: React.KeyboardEvent) {
        if (this.visibleTabs.length === 0)
            return;

        if (e.altKey && (e.key === "ArrowLeft" || e.key === "ArrowRight")) {
            let nextTabIndex = this.selectedTabIndex + (e.key === "ArrowLeft" ? -1 : 1);
            if (nextTabIndex < 0)
                nextTabIndex = this.visibleTabs.length - 1;
            else if (nextTabIndex >= this.visibleTabs.length)
                nextTabIndex = 0;
            this.selectedTabIndex = nextTabIndex;
            e.preventDefault();
            this.onChangeEmitter.fire();
        }
    }

    private handleSelect(tabIndex: number) {
        this.selectedTabIndex = tabIndex;
        this.onChangeEmitter.fire();
    }

}
