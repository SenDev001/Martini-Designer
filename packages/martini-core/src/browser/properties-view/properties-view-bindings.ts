import { bindViewContribution, WidgetFactory } from "@theia/core/lib/browser";
import { interfaces } from "inversify";
import { PropertiesViewContribution } from "./properties-view-contribution";
import { PropertiesViewWidget } from "./properties-view-widget";
import { TabbedPropertiesViewPage } from "./tabbed-properties-view-page";

export const bindPropertiesViewBindings = (bind: interfaces.Bind) => {
    bindViewContribution(bind, PropertiesViewContribution);
    bind(PropertiesViewWidget).toSelf();
    bind(WidgetFactory).toDynamicValue(ctx => ({
        id: PropertiesViewWidget.ID,
        createWidget: () => ctx.container.get<PropertiesViewWidget>(PropertiesViewWidget)
    }));
    bind(TabbedPropertiesViewPage)
        .toSelf()
        .inRequestScope();
};
