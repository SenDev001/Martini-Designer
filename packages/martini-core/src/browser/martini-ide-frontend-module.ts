import { CommandContribution, ILogger, LogLevel, MenuContribution } from "@theia/core";
import {
    FrontendApplicationContribution,
    KeybindingContribution,
    WebSocketConnectionProvider
} from "@theia/core/lib/browser";
import { ContainerModule, inject, injectable, interfaces } from "inversify";
import "../../src/browser/style/index.css";
import { MartiniAccountManager, martiniAccountManagerPath } from "../common/instance/martini-account-manager";
import {
    MartiniInstanceInfoManager,
    martiniInstanceInfoManagerPath
} from "../common/instance/martini-instance-info-manager";
import { ResourceSearchService, resourceSearchServicePath } from "../common/search/resource-search-service";
import { ServiceSearchService, serviceSearchServicePath } from "../common/search/service-search-service";
import { DocumentTypeManager, documentTypeManagerPath } from "../common/tracker/document-type-manager";
import { ContentAssistContributionManager } from "./content-assist/content-assist-contribution";
import { bindDatabaseConnectionBindings } from "./database-connection/database-connection-bindings";
import { ApiResponseHandler } from "./dialogs/api-response-handler";
import { LocalDnDTransfer } from "./dnd/local-dnd-transfer";
import { bindEmbeddedTextEditor } from "./editor/embedded-text-editor";
import { bindMarkdownEditorDialog } from "./editor/md/markdown-editor";
import { bindEndpointBindings } from "./endpoint/endpoint-bindings";
import { bindFilesystemBindings } from "./fs/filesystem-bindings";
import { bindJmsBindings } from "./jms/jms-bindings";
import {
    MartiniIdeCommandContribution,
    MartiniIdeKeybindingContribution,
    MartiniMenuContribution
} from "./martini-ide-contribution";
import { bindNavigatorBindings } from "./navigator/navigator-bindings";
import { bindMartiniPackageBindings } from "./package/martini-package-bindings";
import { DefaultProgressService, ProgressService } from "./progress/progress-service";
import { bindPropertiesViewBindings } from "./properties-view/properties-view-bindings";
import { bindResourceSearchBindings } from "./search/resource-search-bindings";
import { bindWizardBindings } from "./wizard/wizard-bindings";
import { initYupExt } from "./yup/yup-ext";
import { initYupLocale } from "./yup/yup-locale";

export const configureModule: interfaces.ContainerModuleCallBack = (bind, unbind, isBound, rebind) => {
    bindPropertiesViewBindings(bind);
    bindNavigatorBindings(bind);
    bindMartiniPackageBindings(bind);
    bindEndpointBindings(bind);
    bindDatabaseConnectionBindings(bind);
    bindFilesystemBindings(bind);
    bindWizardBindings(bind);
    bindEmbeddedTextEditor(bind, rebind);
    bindMarkdownEditorDialog(bind);
    bindJmsBindings(bind);
    bind(MenuContribution)
        .to(MartiniMenuContribution)
        .inSingletonScope();
    bind(CommandContribution)
        .to(MartiniIdeCommandContribution)
        .inSingletonScope();
    bind(KeybindingContribution)
        .to(MartiniIdeKeybindingContribution)
        .inSingletonScope();
    bind(FrontendApplicationContribution).to(MartiniIdeFrontendContribution);
    bind(ProgressService)
        .to(DefaultProgressService)
        .inSingletonScope();
    bind(MartiniInstanceInfoManager)
        .toDynamicValue(ctx => {
            const connection = ctx.container.get(WebSocketConnectionProvider);
            return connection.createProxy<MartiniInstanceInfoManager>(martiniInstanceInfoManagerPath);
        })
        .inSingletonScope();
    bind(MartiniAccountManager)
        .toDynamicValue(ctx => {
            const connection = ctx.container.get(WebSocketConnectionProvider);
            return connection.createProxy<MartiniAccountManager>(martiniAccountManagerPath);
        })
        .inSingletonScope();
    bind(DocumentTypeManager)
        .toDynamicValue(ctx => {
            const connection = ctx.container.get(WebSocketConnectionProvider);
            return connection.createProxy(documentTypeManagerPath);
        })
        .inSingletonScope();
    bind(ResourceSearchService)
        .toDynamicValue(ctx => {
            const connection = ctx.container.get(WebSocketConnectionProvider);
            return connection.createProxy(resourceSearchServicePath);
        })
        .inSingletonScope();
    bind(ServiceSearchService)
        .toDynamicValue(ctx => {
            const connection = ctx.container.get(WebSocketConnectionProvider);
            return connection.createProxy(serviceSearchServicePath);
        })
        .inSingletonScope();
    bind(ApiResponseHandler)
        .toSelf()
        .inSingletonScope();
    bind(LocalDnDTransfer).toConstantValue(LocalDnDTransfer.INSTANCE);
    bindResourceSearchBindings(bind);
    bind(ContentAssistContributionManager)
        .toSelf()
        .inSingletonScope();
};

export default new ContainerModule(configureModule);

@injectable()
export class MartiniIdeFrontendContribution implements FrontendApplicationContribution {
    @inject(MartiniInstanceInfoManager)
    instanceInfoManager: MartiniInstanceInfoManager;
    @inject(MartiniAccountManager)
    accountManager: MartiniAccountManager;
    @inject(ILogger)
    private readonly logger: ILogger;

    initialize() {
        this.logger.setLogLevel(LogLevel.INFO);
        initYupExt();
        initYupLocale();
    }
}
