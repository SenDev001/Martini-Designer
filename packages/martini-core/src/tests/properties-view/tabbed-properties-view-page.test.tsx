require("reflect-metadata");
import { Widget } from "@theia/core/lib/browser";
import { Container } from "inversify";
import * as React from "react";
import * as Renderer from "react-test-renderer";
import { AbstractPropertiesViewPage } from "../../browser/properties-view/properties-view-page";
import { PropertiesViewTab } from "../../browser/properties-view/properties-view-tab";
import { TabbedPropertiesViewPage } from "../../browser/properties-view/tabbed-properties-view-page";

class DummyTab extends AbstractPropertiesViewPage implements PropertiesViewTab {
    name = "Dummy";
    order = 0;
    visible = true;

    protected doRender(): React.ReactNode {
        return <div>{this.name}</div>;
    }

    isVisible(selection: object): boolean {
        return this.visible;
    }

    onSelectionChange(selection: object) {
        // no-op
    }
}

const tab1 = new DummyTab();
tab1.name = "Tab 1";
tab1.order = 0;
const tab2 = new DummyTab();
tab1.name = "Tab 2";
tab1.order = 1;

const container = new Container();
container.bind(PropertiesViewTab).toConstantValue(tab1);
container.bind(PropertiesViewTab).toConstantValue(tab2);
container.bind(TabbedPropertiesViewPage).toSelf();

test("Should render visible tabs in the right other", () => {
    tab1.visible = true;
    tab2.visible = true;
    const page = container.get(TabbedPropertiesViewPage);
    page.onSelectionChange(new Widget(), {});
    const component = Renderer.create(page.render() as React.ReactElement);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Should not render not visible tabs", () => {
    tab1.visible = false;
    tab2.visible = true;
    const page = container.get(TabbedPropertiesViewPage);
    page.onSelectionChange(new Widget(), {});
    const component = Renderer.create(page.render() as React.ReactElement);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Should default to first tab if current tab index does not exist", () => {
    tab1.visible = true;
    tab2.visible = false;

    const page = container.get(TabbedPropertiesViewPage);
    Object.assign(page, { selectedTabIndex: 1 });
    page.onSelectionChange(new Widget(), {});

    const component = Renderer.create(page.render() as React.ReactElement);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
