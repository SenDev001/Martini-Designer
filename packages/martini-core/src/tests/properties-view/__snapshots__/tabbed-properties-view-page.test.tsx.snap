// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should default to first tab if current tab index does not exist 1`] = `
<div
  className="react-tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <div
    aria-labelledby="react-tabs-6"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-7"
    role="tabpanel"
    style={
      Object {
        "flex": 1,
        "margin": 5,
        "overflowY": "auto",
      }
    }
  >
    <div
      className="sc-AxjAm btPfNO"
    >
      <div
        className="content"
      >
        <div>
          Tab 2
        </div>
      </div>
    </div>
  </div>
  <ul
    className="react-tabs__tab-list"
    role="tablist"
    style={
      Object {
        "flexShrink": 0,
      }
    }
  >
    <li
      aria-controls="react-tabs-7"
      aria-disabled="false"
      aria-selected="true"
      className="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-6"
      role="tab"
      tabIndex="0"
    >
      Tab 2
    </li>
  </ul>
</div>
`;

exports[`Should not render not visible tabs 1`] = `
<div
  className="react-tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <div
    aria-labelledby="react-tabs-4"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-5"
    role="tabpanel"
    style={
      Object {
        "flex": 1,
        "margin": 5,
        "overflowY": "auto",
      }
    }
  >
    <div
      className="sc-AxjAm btPfNO"
    >
      <div
        className="content"
      >
        <div>
          Dummy
        </div>
      </div>
    </div>
  </div>
  <ul
    className="react-tabs__tab-list"
    role="tablist"
    style={
      Object {
        "flexShrink": 0,
      }
    }
  >
    <li
      aria-controls="react-tabs-5"
      aria-disabled="false"
      aria-selected="true"
      className="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-4"
      role="tab"
      tabIndex="0"
    >
      Dummy
    </li>
  </ul>
</div>
`;

exports[`Should render visible tabs in the right other 1`] = `
<div
  className="react-tabs"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
    }
  }
>
  <div
    aria-labelledby="react-tabs-0"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
    style={
      Object {
        "flex": 1,
        "margin": 5,
        "overflowY": "auto",
      }
    }
  >
    <div
      className="sc-AxjAm btPfNO"
    >
      <div
        className="content"
      >
        <div>
          Dummy
        </div>
      </div>
    </div>
  </div>
  <div
    aria-labelledby="react-tabs-2"
    className="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
    style={
      Object {
        "flex": 1,
        "margin": 5,
        "overflowY": "auto",
      }
    }
  >
    <div
      className="sc-AxjAm btPfNO"
    >
      <div
        className="content"
      >
        <div>
          Tab 2
        </div>
      </div>
    </div>
  </div>
  <ul
    className="react-tabs__tab-list"
    role="tablist"
    style={
      Object {
        "flexShrink": 0,
      }
    }
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      className="react-tabs__tab react-tabs__tab--selected"
      id="react-tabs-0"
      role="tab"
      tabIndex="0"
    >
      Dummy
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      className="react-tabs__tab"
      id="react-tabs-2"
      role="tab"
      tabIndex={null}
    >
      Tab 2
    </li>
  </ul>
</div>
`;
